### PR TITLE
Added lf-ques-im-hist-att model

### PR DESCRIPTION
**Pull request by interritus1996**
Simple attention mechanism over image using question features has been added to Late Fusion model in lf-ques-im-hist-att.lua and corresponding changes have been done in other files to accommodate 14x14x512 feature vectors.

### CODE REVIEW

**Comment by abhshkdz** [comment]
There are merge conflicts with https://github.com/batra-mlp-lab/visdial/commit/ea9dff014e0f5e002976ec020b2e05375b3adf24.
1. It's likely that changes in `model.lua`, `dataloader.lua` and `opts.lua` should already be in.
2. [`encoders/lf-ques-im-hist-att.lua`](https://github.com/interritus1996/visdial/blob/master/encoders/lf-ques-im-hist-att.lua) looks good. (Nitpick: Let's rename it to `lf-att-ques-im-hist.lua`)
3. I like the updates to [`data/prepro_img.lua`](https://github.com/interritus1996/visdial/blob/master/data/prepro_img.lua), MUCH better than keeping 2 separate preprocessing scripts as there are now. Let's keep those. Good job! 

I can merge once you rebase on master and resolve these.

How are the accuracies for `lf-ques-im-hist-att`?

**Comment by interritus1996** [comment]
I have made the changes as asked.

Due to my limited computational resources, I was not able to train the model on all 80,000 images; instead, I selected a small set of images (500 images) and tried to minimise the loss over these images.

**Comment by abhshkdz** [comment]
Apologies for the delay, finally got around to benchmarking `lf-att-ques-im-hist` and `lf-att-ques-im-hist-san`. I would've merged this earlier if you'd reported results as well.

`lf-att-ques-im-hist` seems buggy, I was consistently getting worse performance than `lf-ques-im-hist`.

`lf-att-ques-im-hist-san` looks good. It gets 0.5305 MRR, 68.38 R@10. Renamed it to `lf-att-ques-im-hist` and merged in https://github.com/batra-mlp-lab/visdial/commit/a34094c5d4979f8c2c204e766eae3665b0d342a2.

Thanks for the PR! 👍
